### PR TITLE
Feat/desired lanes

### PR DESCRIPTION
**Pull request by AmirBandarii**

### CODE REVIEW

File: src/libs/entities/desired-lanes.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  @IsString()
+  fromCity: string;
+
+  @Column({ type: 'varchar' })
+  @IsString()
+  toCity: string;
+
+  @Column({ type: 'varchar' })
+  @IsString()
+  customerInfoUuid: string;
```
Comment (harutyunyan611) [inline src/libs/entities/desired-lanes.ts]: Why not id?

File: src/public/js/script.js [hunk 2]
```diff
@@ -145,21 +154,20 @@ function hideShowDiv(val) {
...
-document.getElementById('submit').addEventListener('click', function (event) {
+document.getElementById('submit').addEventListener('click', async function(event) {
   event.preventDefault()
   if (!form.checkValidity()) {
     event.preventDefault()
     event.stopPropagation()
     form.classList.add('was-validated');
   } else {
-    postData()
+    uniqueUuid = generateUniqueString(10)
```
Comment (harutyunyan611) [inline src/public/js/script.js]: why on frontend?

File: src/public/js/script.js [hunk 3]
```diff
@@ -145,21 +154,20 @@ function hideShowDiv(val) {
...
+    await postData()
+    await addDesiredLanes()
```
Comment (harutyunyan611) [inline src/public/js/script.js]: Why not with the same request? What if second one will fail?

File: src/public/js/script.js [hunk 4]
```diff
@@ -115,25 +117,32 @@ async function postData() {
...
+async function addDesiredLanes(){
+  const fromCityOptionalInput = document.getElementById('fromCity').value
+  const toCityOptionalInput = document.getElementById('toCity').value
+  try {
+    const response = await axios.post('http://localhost:80/api/desiredLanes', [{
+      fromCity : fromCityOptionalInput,
+      toCity : toCityOptionalInput,
+      customerInfoUuid : uniqueUuid,
+    }])
+    return await response
```
Comment (harutyunyan611) [inline src/public/js/script.js]: This cannot not work, in array it always have one element

Comment (harutyunyan611) [reply src/public/js/script.js]: Relation should not be decided from where, btw will not work

File: src/public/js/script.js [hunk 5]
```diff
@@ -176,21 +184,46 @@ function handleRadioClick() {
...
+            </div>
+        </div>
+            `;
+  document.querySelector(".delivery-address").appendChild(newInputFields);
+});
+document.addEventListener("click", function(event) {
+  if (event.target.id === "remove") {
+    removeInputFields(event);
+  }
+});
```
Comment (harutyunyan611) [inline src/public/js/script.js]: Why we're adding event listener to whole document? Id should be unique accross document

File: src/routes/customer-routes.ts [hunk 6]
```diff
@@ -6,3 +6,4 @@ export const router = express.Router()
 
 router.post('/upload', uploadCoi.single('file'), CustomerController.uploadPhoto)
 router.post('/company', CustomerController.createCustomerInfo)
+router.post('/desiredLanes', CustomerController.desiredLanesCreate)
```
Comment (harutyunyan611) [inline src/routes/customer-routes.ts]: do not use camelcase inside endpoints, btw this endpoint should be removed

File: src/services/desired-lanes.service.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  async createDesiredLanes(desiredLanes : DesiredLanesDto[]) : Promise<InsertResult[]> {
+    const insertResults: InsertResult[] = [];
+    desiredLanes.map(async (lane) => {
+      const result = await this.desiredLanesRepository.insert({
+        id: lane.id,
+        fromCity: lane.fromCity,
+        toCity: lane.toCity,
+        customerInfoUuid: lane.customerInfoUuid
+      });
```
Comment (harutyunyan611) [inline src/services/desired-lanes.service.ts]: Can we bulk insert?

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

File: src/libs/entities/desired-lanes.ts (hunk 1 above)
Comment (AmirBandarii) [reply src/libs/entities/desired-lanes.ts]: done

File: src/public/js/script.js (hunk 2 above)
Comment (AmirBandarii) [reply src/public/js/script.js]: done

File: src/public/js/script.js (hunk 3 above)
Comment (AmirBandarii) [reply src/public/js/script.js]: done

File: src/public/js/script.js (hunk 4 above)
Comment (AmirBandarii) [reply src/public/js/script.js]: done

File: src/public/js/script.js (hunk 5 above)
Comment (AmirBandarii) [reply src/public/js/script.js]: done

File: src/routes/customer-routes.ts (hunk 6 above)
Comment (AmirBandarii) [reply src/routes/customer-routes.ts]: done

File: src/services/desired-lanes.service.ts (hunk 7 above)
Comment (AmirBandarii) [reply src/services/desired-lanes.service.ts]: done

File: src/controller/customer.controller.ts [hunk 8]
```diff
@@ -13,17 +13,18 @@ class CustomerController {
...
+
+
   async createCustomerInfo(req: Request, res: Response): Promise<unknown> {
     try {
-      const customerService = CustomerInfoService.createCustomerInfo(req.body)
-
-      return res.json(customerService)
+      const customerInfoId = await CustomerInfoService.createCustomerInfo(req.body.customerInfo);
+      const desiredLanes = await DesiredLanesService.createDesiredLanes(customerInfoId, req.body.desiredLanes)
+      return res.json({desiredLanes})
```
Comment (harutyunyan611) [inline src/controller/customer.controller.ts:25]: Why you're sending it to frontend?

File: src/libs/db/db.ts [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
+    type :"mysql",
+    //database: "dbName",
+    host: process.env.DB_HOST,
+    username: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_NAME,
+    port: parseInt(process.env.DB_PORT || '3306'),
+    entities: [CustomerInfo,DesiredLanes],
+    synchronize: false,
 })
```
Comment (harutyunyan611) [inline src/libs/db/db.ts:18]: Please bring back my `sqlite` 😁

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

**Comment by harutyunyan611** [comment]
Merge yourself once addressed this comments

**Pull request merged**